### PR TITLE
Project initialization

### PR DESCRIPTION
**Pull request by imrankhan-stixor**
# Description

Initialized the project created the basic layout of the project, added the routing, made a basic navbar to the project which will be updated later on, added the sign up form and some other useful components

## Type of change

Please select type of issue addressed,

- [ ] Bug fix (None)

- [ ] New feature ( None) 

# Dependences / Libraries Installed

Please list the packages and libraries installed (if any)

- [x] Yup  ( https://www.npmjs.com/package/yup ) added this for validations and stuff.
- [x] Redix Ui (added by Shad cn)

# How Has This Been Tested By The Developer?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Tested the components and things to match the ui

- [x] Tested the responsiveness

# Checklist:

- [x] I have performed a self review of my code.

- [x] I have removed consoles, debuggers and unnecessary code blocks.

- [x] My code follows the style guidelines of this project

- [x] My changes generate no new warnings or errors

- [ ] I have commented my code, particularly in hard-to-understand areas

- [ ] New and existing unit tests pass locally with my changes

- [x] Any dependent changes have been merged and published in downstream modules

- [x] I understand that I am accountable for any code added in this PR, and any negative impact will effect my performance evaluation

### CODE REVIEW

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: use rems

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: remove this if not required

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: remove this extra div

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: here too

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: remove this extra div

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: use rems instead of px

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: by default font is normal

Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: instead of importing `inter` font, use it from tailwind config

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: use rems

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: rems

**Comment by sananbintahir** [comment]
@hassantahir767 can you review this please?

@imrankhan-stixor I don't see any UI-related things on sign-up. Can you please properly list down what you've done in this PR so that its easier to review? I see buggy UI on the netlify deployed version.

**Comment by sananbintahir** [comment]
I want better code quality standards on this so I've added a rule because of which 2 reviewers need to approve before a branch can be merged. @fawadmehmood1 you should review this too.

**Comment by imrankhan-stixor** [comment]
> @hassantahir767 can you review this please?
> 
> @imrankhan-stixor I don't see any UI-related things on sign-up. Can you please properly list down what you've done in this PR so that its easier to review? I see buggy UI on the netlify deployed version.



> @hassantahir767 can you review this please?
> 
> @imrankhan-stixor I don't see any UI-related things on sign-up. Can you please properly list down what you've done in this PR so that its easier to review? I see buggy UI on the netlify deployed version.

@sananbintahir mostly it has the components, fonts, colors and stuff like that and some of the code for the sign up, main reason for merging is so that develop gets updated and @fawadmehmood1 can also start working from the current code

**Comment by sananbintahir** [comment]
> 
> @sananbintahir mostly it has the components, fonts, colors and stuff like that and some of the code for the sign up, main reason for merging is so that develop gets updated and @fawadmehmood1 can also start working from the current code

I understand, but you've done more than was asked. This also includes routing, navigation etc. Project-Level code as well as UI and component level code. I want you to list down exactly what was done and how it can be tested (e.g. sign in page on /sign-in, responsive navbar on /apply)

I say this because you've done some weird things that if they get merged will never get changed. This includes missing hover effects on the left navbar, as well as weird routing, where there is a significant delay when routing between pages (usually its instantaneous on nextjs)

image

File: app/types.ts
Comment (hassantahir767) [inline app/types.ts:1]: @imrankhan-stixor  The types file has a lot of unnecessary types defined. ( seems to be from Supplier app ). Please remove all of them and for now just keep those which makes sense and are actually required.

File: components/side-bar.tsx
Comment (hassantahir767) [inline components/side-bar.tsx]: use `p-2` instead

File: components/side-bar.tsx
Comment (hassantahir767) [inline components/side-bar.tsx]: use `p-2 `instead

File: components/side-bar.tsx
Comment (hassantahir767) [inline components/side-bar.tsx]: use `h-px` instead

File: components/side-drawer.tsx
Comment (hassantahir767) [inline components/side-drawer.tsx]: use rem instead for defining with.

File: app/(auth)/sign-up/page.tsx
Comment (hassantahir767) [inline app/(auth)/sign-up/page.tsx]: gap cannot be constant, can cause trouble in responsiveness. use rem instead

File: app/(auth)/sign-up/page.tsx
Comment (hassantahir767) [inline app/(auth)/sign-up/page.tsx]: use rem instead for defining the widths

File: app/(auth)/sign-up/page.tsx
Comment (hassantahir767) [inline app/(auth)/sign-up/page.tsx]: use `w-full` instead. it does the same. No need for w-[100%]

File: app/(auth)/sign-up/page.tsx
Comment (hassantahir767) [inline app/(auth)/sign-up/page.tsx]: use rem instead

File: app/(auth)/sign-up/page.tsx
Comment (hassantahir767) [inline app/(auth)/sign-up/page.tsx]: `h-px`

File: app/(auth)/sign-up/page.tsx
Comment (hassantahir767) [inline app/(auth)/sign-up/page.tsx]: use rem for gap

File: app/(auth)/sign-up/page.tsx
Comment (hassantahir767) [inline app/(auth)/sign-up/page.tsx]: define color in tailwind config.

File: app/types.ts
Comment (fawadmehmood1) [inline app/types.ts]: remove these types, not relevant to this project

File: app/(panel)/apply/page.tsx
Comment (hassantahir767) [inline app/(panel)/apply/page.tsx]: Define pages like this instead

`import { NextPage } from "next";

const Apply: NextPage = () => {`

**Comment by hassantahir767** [comment]
@imrankhan-stixor  General comment. Please use rem specially for font sizes, gaps and paddings. Using pixels would make it a constant value across all dimensions and resolutions.

File: components/side-bar.tsx
Comment (fawadmehmood1) [inline components/side-bar.tsx]: use next link here instead of div

File: components/side-bar.tsx
Comment (fawadmehmood1) [inline components/side-bar.tsx:49]: use next link here instead of div

File: components/side-bar.tsx
Comment (fawadmehmood1) [inline components/side-bar.tsx]: there isn't any expanded menu, you can remove this

File: components/side-bar.tsx
Comment (fawadmehmood1) [inline components/side-bar.tsx]: instead of onClick, wrap image inside next link, expandedMenuIndex is not needed

File: app/(auth)/sign-up/page.tsx
Comment (imrankhan-stixor) [reply app/(auth)/sign-up/page.tsx]: It's just used one time over here

File: components/side-bar.tsx
Comment (fawadmehmood1) [inline components/side-bar.tsx]: this is also not required, use `pathname` for making link active.

File: components/side-bar.tsx
Comment (fawadmehmood1) [inline components/side-bar.tsx]: not required

File: components/side-bar.tsx
Comment (fawadmehmood1) [inline components/side-bar.tsx]: use color from tailwind config

File: app/(auth)/sign-up/page.tsx
Comment (sananbintahir) [reply app/(auth)/sign-up/page.tsx]: @imrankhan-stixor there shouldn't be any unique color. If this is a 1-off, bring this to Waseeq's attention so that he can fix it in the design. Every color in the website should be properly defined on the FIGMA and be initialized in the tailwind config.

Please not that we will also add a feature later on down the line where people can customize theme colors. This will be an issue at that point too.

File: components/side-bar.tsx
Comment (fawadmehmood1) [inline components/side-bar.tsx]: use pathname

File: components/side-bar.tsx
Comment (fawadmehmood1) [inline components/side-bar.tsx]: pathname

**Comment by sananbintahir** [comment]
Also @imrankhan-stixor I think waseeq is using radix icons, please confirm from him once.

File: components/ui/form-page-template.tsx
Comment (fawadmehmood1) [inline components/ui/form-page-template.tsx]: use rems

File: components/ui/form-page-template.tsx
Comment (fawadmehmood1) [inline components/ui/form-page-template.tsx]: use rems

File: components/ui/header.tsx
Comment (fawadmehmood1) [inline components/ui/header.tsx]: you adjust the size by using css

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: use `lucide-react` for icons

File: formik/initial-values/index.ts
Comment (fawadmehmood1) [inline formik/initial-values/index.ts]: use separate file for form interfaces

File: components/side-bar.tsx
Comment (fawadmehmood1) [inline components/side-bar.tsx]: use `lucide` icons here and change colors, instead of using svg's and replacing them with coloured svg's. Also delete these svg's

File: components/ui/form-page-template.tsx
Comment (fawadmehmood1) [inline components/ui/form-page-template.tsx]: use color from tailwind config

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: these are buttons, use proper button here

File: package.json
Comment (fawadmehmood1) [inline package.json]: remove this

File: routes.ts
Comment (fawadmehmood1) [inline routes.ts]: remove commented code

File: components/formik/input-field.tsx
Comment (fawadmehmood1) [inline components/formik/input-field.tsx]: why are we not using shadcn input component ?

File: components/formik/input-field.tsx
Comment (fawadmehmood1) [inline components/formik/input-field.tsx]: use `react-lucide` icons

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: we should `shadcn` form and fields

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx]: use this font from tailwind config

File: formik/forms/index.ts
Comment (fawadmehmood1) [inline formik/forms/index.ts]: implement a proper file structure for forms, as there will be many forms in the future.

**Review by fawadmehmood1 — CHANGES_REQUESTED** [review]

**Comment by fawadmehmood1** [comment]
@imrankhan-stixor, 
- Replace pixel values with rems
- Use `react-lucide` for icons, you'll find all the icons used in the design there (removes SVGs). 
- Consider using  shadcn components for forms
- Use colors from tailwind config file

File: app/(auth)/sign-up/page.tsx
Comment (imrankhan-stixor) [reply app/(auth)/sign-up/page.tsx]: ok sanan bhai basically i used this color for jira's icon, but will define the colors in the tailwind

File: app/(auth)/sign-up/page.tsx
Comment (fawadmehmood1) [inline app/(auth)/sign-up/page.tsx:121]: remove this is not required

File: components/side-bar.tsx
Comment (fawadmehmood1) [inline components/side-bar.tsx]: you are using same width for screens after `md`, no need to add `${!inSideDrawer && "lg:w-[15rem] lg:min-w-[15rem]"}`
`min-w` and `width` are some, you can remove `min-w`

File: components/side-bar.tsx
Comment (fawadmehmood1) [inline components/side-bar.tsx]: text color for `hr` tag ? 
It will be `bg`

**Review by fawadmehmood1 — CHANGES_REQUESTED** [review]

**Review by hassantahir767 — DISMISSED** [review]
@fawadmehmood1  Merge after your review.

File: app/(auth)/sign-up/page.tsx
Comment (imrankhan-stixor) [reply app/(auth)/sign-up/page.tsx:121]: It'll be used when the api is made

File: components/side-bar.tsx
Comment (imrankhan-stixor) [reply components/side-bar.tsx:77]: the line gets the text color not the bg color

Comment (imrankhan-stixor) [reply components/side-bar.tsx]: removed it

**Review by fawadmehmood1 — APPROVED** [review]

**Review by hassantahir767 — APPROVED** [review]

**Pull request merged**
